hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java

```diff
@@ -135,30 +137,15 @@ public CallOutcome call() {
             }
             return CallOutcome.fromResultsWithMaybeSidecars(
                     result.asProtoResultOf(ethTxDataIfApplicable(), rootProxyWorldUpdater), result);
-        } catch (AbortException e) {
-            if (e.isChargeable() && contractsConfig.chargeGasOnPreEvmException()) {
-                gasCharging.chargeGasForAbortedTransaction(
-                        requireNonNull(e.senderId()), hederaEvmContext, rootProxyWorldUpdater, hevmTransaction);
-            }
-            // Commit any HAPI fees that were charged before aborting
-            rootProxyWorldUpdater.commit();
-
-            ContractID recipientId = null;
-            if (!INVALID_CONTRACT_ID.equals(e.getStatus())) {
-                recipientId = hevmTransaction.contractId();
-            }
-
-            var result = HederaEvmTransactionResult.fromAborted(e.senderId(), recipientId, e.getStatus());
-
-            if (context.body().hasEthereumTransaction()) {
-                final var sender = rootProxyWorldUpdater.getHederaAccount(e.senderId());
-                if (sender != null) {
-                    result = result.withSignerNonce(sender.getNonce());
-                }
-            }
-
-            return CallOutcome.fromResultsWithoutSidecars(
-                    result.asProtoResultOf(ethTxDataIfApplicable(), rootProxyWorldUpdater), result);
+        } catch (HandleException e) {
+            final var sender = rootProxyWorldUpdater.getHederaAccount(hevmTransaction.senderId());
+            final var senderId = sender != null ? sender.hederaId() : hevmTransaction.senderId();
+
+            return maybeChargeFeesAndReturnOutcome(
+                    hevmTransaction.withException(e),
+                    senderId,
+                    sender,
+                    hevmTransaction.isContractCall() && contractsConfig.chargeGasOnPreEvmException());
```

Comment:
I'm not sure why a config named `chargeGasOnPreEvmException` controls this (emphasis on the **`Pre`**), since this handler encloses `processTransaction` where the EVM is run.

(That gate is up at L113 where I'd expect it to be.)

Reply:
The config's purpose may have evolved since it was first thought of.  I will changed the name to `chargeGasOnEvmHandleException`